core/src/main/java/org/sonarsource/sonarlint/core/websocket/WebSocketService.java

```diff
@@ -61,6 +66,22 @@ public WebSocketService(SonarLintClient client, ConnectionConfigurationRepositor
       .dispatch(QualityGateChangedEvent.class, new ShowSmartNotificationOnQualityGateChangedEvent(client, configurationRepository));
   }
 
+  @PostConstruct
+  public void initialize() {
+    ScheduledExecutorService webSocketConnectionRefresher = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, "WebSocket connection refresher"));
```

Comment:
You might need to store this in a field and properly shut it down when the service is shutdown.